pyomo/core/base/plugin.py

```diff
@@ -350,7 +351,17 @@ def _apply_to(self, model, **kwds):
             "The Transformation.apply_to method is not implemented.")
 
     def _create_using(self, model, **kwds):
+        # Put all the kwds onto the model so that when we clone the
+        # model any references to things on the model are correctly
+        # updated to point to the new instance.  Note that users &
+        # transformation developers cannot rely on things happening by
+        # argument side effect.
```

Comment:
Should we add in a warning/disclaimer about functions being passed in to kwds?

Reply:
@qtothec: I don't think that is an issue.  Functions generally don't store state and therefore shouldn't care which instance they may eventually be called on, and functors (callable objects), which was my original worry turn out to work fine.  The only case that might burn users are functions that refer to the model directly without passing it in as an argument.  That will fail, but I also don't think there is any way that we could catch that edge case.